test/unit_test.ts

```diff
@@ -3,245 +3,178 @@ import chai = require('chai');
-  });
-
-  it('should include constructors', () => {
-    check(
-        `
-      export class A {
-        constructor(a:string) {}
+describe('unit test', () => {
+  it('should support imports without capturing imports', () => {
+    const input = `
+      import {A} from './classes_and_interfaces';
+      export declare class C {
+          field: A;
       }
-    `,
-        ['A', 'A.constructor(a:string)']);
-  });
-
-  it('should support interfaces', () => {
-    check(
-        `
-      export interface A {
-        field:string;
-        method(a:string):number;
+    `;
+    const expected = `
+      export declare class C {
+          field: A;
       }
-    `,
-        ['A', 'A.field:string', 'A.method(a:string):number']);
+    `;
+    check({'classes_and_interfaces.d.ts': classesAndInterfaces, 'file.d.ts': input}, expected);
   });
 
-  it('should support generics', () => {
-    check(
-        `
-      export class A<T> {
-        field:T;
-        method(a:T):T { return null; }
+  it('should support imports with prefixes without capturing imports', () => {
+    const input = `
+      import * as t from './classes_and_interfaces';
+      export declare class C {
+          field: t.A;
       }
-    `,
-        ['A<T>', 'A.field:T', 'A.method(a:T):T']);
-  });
-
-  it('should support static members', () => {
-    check(
-        `
-      export class A {
-        static field: string;
-        static method(a: string): number {}
+    `;
+    const expected = `
+      export declare class C {
+          field: t.A;
       }
-    `,
-        ['A', 'A.field:string', 'A.method(a:string):number']);
+    `;
+    check({'classes_and_interfaces.d.ts': classesAndInterfaces, 'file.d.ts': input}, expected);
   });
 
-  it('should support arrays', () => {
-    check(
-        `
-      export var a: Array<Array<string>>;
-      export var b: string[][];
-    `,
-        ['var a:Array<Array<string>>', 'var b:string[][]']);
+  it('should throw on aliased reexports', () => {
+    const input = `export { A as Apple } from './classes_and_interfaces';`;
+    checkThrows(
+        {'classes_and_interfaces.d.ts': classesAndInterfaces, 'file.d.ts': input},
+        'Symbol "A" was aliased as "Apple"');
   });
 
-  it('should support map', () => {
-    check(
-        `
-      export var a: Map<Map<string, number>, number>;
-    `,
-        ['var a:Map<Map<string, number>, number>']);
+  // FIXME
+  it('should remove reexported external symbols', () => {
+    const input = `
+      export { Foo } from 'some-external-module-that-cannot-be-resolved';
+    `;
+    const expected = `
+    `;
+    check({'classes_and_interfaces.d.ts': classesAndInterfaces, 'file.d.ts': input}, expected);
   });
 
-  it('should support getters and setters', () => {
-    check(
-        `
-      export class A {
-        get a(): string {}
-        set a(v:string){}
-        get b() {}
-        set b(v) {}
+  it('should sort exports', () => {
```

Comment:
does this sorting happen across all the files when we are doing multi file reexport?